File: src/plugin/item/SummoningPouchItemPlugin.java
package plugin.item;

import org.redrune.game.content.entity.actor.player.skills.summoning.Summoning;
import org.redrune.game.content.entity.actor.player.skills.summoning.Summoning.Pouches;
import org.redrune.game.content.plugin.type.ItemPlugin;
import org.redrune.game.entity.actor.player.Player;
import org.redrune.game.entity.item.Item;

import java.util.Arrays;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 2019-02-08
 */
public class SummoningPouchItemPlugin implements ItemPlugin {
	
	@Override
	public boolean handle(Player player, Item item, int slotId, String option) {
		Pouches pouch = Pouches.forId(item.getId());
		if (pouch == null) {
			return false;
		}
		Summoning.spawnFamiliar(player, pouch);
		return true;
	}
	
	@Override
	public void register() {
		Arrays.stream(Pouches.values()).forEach(pouch -> registerItem(pouch.getPouchId(), "Summon"));
	}
}

File: src/main/java/plugin/item/SummoningPouchItemPlugin.kt
package plugin.item

import game.content.entity.actor.player.skills.summoning.Summoning
import game.content.plugin.type.ItemPlugin
import game.entity.actor.player.Player
import game.entity.item.Item
import java.util.*

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 2019-02-08
 */
class SummoningPouchItemPlugin : ItemPlugin {
    override fun handle(player: Player, item: Item, slotId: Int, option: String): Boolean {
        val pouch: Summoning.Pouches = Summoning.Pouches.forId(item.id) ?: return false
        Summoning.spawnFamiliar(player, pouch)
        return true
    }

    override fun register() {
        Arrays.stream(Summoning.Pouches.values())
            .forEach { pouch: Summoning.Pouches ->
                registerItem(
                    pouch.pouchId,
                    "Summon"
                )
            }
    }
}